luarules/gadgets/game_disable_assist_ally_construction.lua

```diff
@@ -0,0 +1,64 @@
+function gadget:GetInfo()
+	return {
+		name    = 'Disable Assist Ally Construction',
+		desc    = 'Disable assisting allied units (e.g. labs and units/buildings under construction) when modoption is enabled',
+		author  = 'Rimilel',
+		date    = 'April 2024',
+		license = 'GNU GPL, v2 or later',
+		layer   = 0,
+		enabled = true
+	}
+end
+
+----------------------------------------------------------------
+-- Synced only
+----------------------------------------------------------------
+if not gadgetHandler:IsSyncedCode() then
+	return false
+end
+
+local function isComplete(u)
+    local _,_,_,_,buildProgress=Spring.GetUnitHealth(u)
+    if buildProgress and buildProgress>=1 then
+        return true
+    else
+        return false
+    end
+end
+
+function gadget:AllowCommand(unitID, unitDefID, unitTeam, cmdID, cmdParams, cmdOptions, cmdTag, synced)
+
+    if(not Spring.GetModOptions().disable_assist_ally_construction) then
+        return true
+    end
+
+    -- Disallow guard commands onto other ally's units. (Optional todo: disallow guards on allied build-capable units and labs only)
```

Comment:
Probably something like this. I don't know why spies have build power though. Minelayer though its obvious, it builds mines, but it can't rez.
```local spyDefs = {UnitDefNames.armspy.id, UnitDefNames.corspy.id}
local function isBuilderType(unitDefID)
	local unitDef = UnitDefs[unitDefID]
	return (unitDef.isBuilder and not unitDef.isFactory and (unitDef.buildOptions and unitDef.canReclaim and not unitDef.canResurrect) and (spyDefs[1] ~= unitDefID) and (spyDefs[2] ~= unitDefID))
end```

Reply:
added in 899e0ae76bc4cf3cbda47b2a6b1433ac33d5c446, thanks

Reply:
Settled with  
```
if #targetUnitDef.buildOptions > 0 or targetUnitDef.canAssist then
				return false
end
```